### PR TITLE
Cleanup the webpack config

### PR DESCRIPTION
**Pull request by elfenlaid**
The changes aim to clean up the webpack config. The aim here is to simplify the transition to esbuild, but also make things more explicit.

I'll go through the changes in review comments, but I'm far from being the Webpack expert. Please take my tweaks with the grain of salt 🙏

### CODE REVIEW

File: assets/package.json [hunk 1]
```diff
@@ -16,7 +16,6 @@
     "phoenix_html": "file:../deps/phoenix_html",
     "phoenix_live_view": "file:../deps/phoenix_live_view",
     "phoenix": "file:../deps/phoenix",
-    "popper.js": "^1.15.0",
```
Comment (elfenlaid) [inline assets/package.json:19]: I could be wrong, but neither the codebase, nor any of its dependencies use `popper.js`

File: assets/yarn.lock [hunk 2]
```diff
@@ -2300,7 +2300,7 @@ path-type@^5.0.0:
   version "1.7.2"
 
 "phoenix_html@file:../deps/phoenix_html":
-  version "3.3.1"
```
Comment (elfenlaid) [inline assets/yarn.lock:2303]: Yarn gets the `phoenix_html` package from the Elixir `deps` directory. Currently, it's of `3.3.3` version. In other words, the change is not related to the `package.json` changes, but I thought that it would be okay to include it here.

File: assets/webpack.config.js [hunk 3]
```diff
@@ -44,35 +44,14 @@ module.exports = (env, options) => ({
         test: /\.(png|woff|woff2|eot|ttf|svg)$/,
         loader: "file-loader",
         options: {outputPath: "css", publicPath: "/css"}
-      },
-      {
```
Comment (elfenlaid) [inline assets/webpack.config.js:48]: Notice that webpack checks for the `/umd` directory. UMD stands for Universal Module Definition. Seems that Bootstrap 3.4.1 doesn't have these module definitions. They were added in version 4, and now in version 5 they are gone. 

Here's the screenshot of the bootstrap `dist` folder:

CleanShot 2024-11-30 at 21 15 20@2x

As of now, it doesn't affect bundled files. Though, we can remove `/umd` postfix to inject jquery here, but it seems a more reliable way of injection is to use `ProvidePlugin`. The plugin is also very close to what `esbuild` does with its injections.

File: assets/webpack.config.js [hunk 4]
```diff
@@ -44,35 +44,14 @@ module.exports = (env, options) => ({
...
-        options: {jQuery: 'jquery'}
       }
     ]
   },
   plugins: [
     new MiniCssExtractPlugin({ filename: 'css/app.css' }),
     new CopyWebpackPlugin({patterns: [{ from: 'static/', to: '.' }]}),
     new webpack.ProvidePlugin({
-      $: "jquery",
       jQuery: "jquery",
```
Comment (elfenlaid) [inline assets/webpack.config.js:54]: Leaving `jQuery: "jquery",` for the bootstrap jquery injection.

File: assets/webpack.config.js [hunk 5]
```diff
@@ -44,35 +44,14 @@ module.exports = (env, options) => ({
...
-      "window.jQuery": "jquery",
```
Comment (elfenlaid) [inline assets/webpack.config.js:61]: As far as I understand, `ProvidePlugin` implicitly imports definitions from this dictionary to bundled js files. There's only one file in the application – `app.js`, and it doesn't use any of these definitions directly.

`app.js` uses some bootstrap functionality, like button tooltips and such, but they are available through `import "bootstrap"`.

I could be wrong, but I think they can be safely removed.

File: assets/package.json (hunk 1 above)
Comment (ericmj) [reply assets/package.json:19]: Could it be used by this or is that another popper implementation?

Screenshot 2024-12-03 at 11 04 55

Comment (ericmj) [reply assets/package.json:19]: It's still working so you are right.

**Review by ericmj — APPROVED** [review]

File: assets/package.json (hunk 1 above)
Comment (elfenlaid) [reply assets/package.json:19]: Yeah, I think it's Bootstrap `tooltip`, similar to what those copy buttons show:

CleanShot 2024-12-03 at 11 23 22@2x

**Comment by ericmj** [comment]
Thank you!

**Comment by elfenlaid** [comment]
Thank you for looking into it! 🙇
